crates/sui-types/src/temporary_store.rs

```diff
@@ -181,7 +186,8 @@ impl<S> TemporaryStore<S> {
     }
 
     /// Break up the structure and return its internal stores (objects, active_inputs, written, deleted)
-    pub fn into_inner(self) -> InnerTemporaryStore {
+    /// TODO: can we make this more efficient e.g by minimizing clones
+    pub fn inner(&self) -> InnerTemporaryStore {
```

Comment:
I think a rough convention in Rust is `into_*` moves & consumes the data.
So since we're no longer consuming `self`, I renamed it to match. I could be wrong about this naming style though.

Reply:
Just getting started here, but this feels really bad that we now have to clone. This basically 2xs the amount of allocations we will be doing at the end of a transaction right? (which includes cloning all of the objects, which might be a lot of data)

Reply:
Clones removed